src/analysis/nextest/nextest_line_analyser.rs

```diff
@@ -48,31 +53,68 @@ impl LineAnalyzer for NextestLineAnalyzer {
     }
 }
-
-fn title_key(content: &TLine) -> Option<String> {
-    title_key_v1(content).or_else(|| title_key_v2(content))
-}
-/// Return the key when the line is like "--- STD(OUT|ERR): somekey ---"
-fn title_key_v1(content: &TLine) -> Option<String> {
-    let mut strings = content.strings.iter();
-    let first = strings.next()?;
-    if !regex_is_match!(r"^--- STD(OUT|ERR):\s*", &first.raw) {
-        return None;
+impl NextestLineAnalyzer {
+    fn stdx_section_key(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        self.stdx_section_key_v1(content)
+            .or_else(|| self.stdx_section_key_v2(content))
+            .or_else(|| self.stdx_section_key_v3(content))
     }
-    extract_key_after_crate_name_v1(strings)
-}
-/// Return the key when the line is like "──── STD(OUT|ERR): cratename some::key"
-fn title_key_v2(content: &TLine) -> Option<String> {
-    let mut strings = content.strings.iter();
-    let ts = strings.next()?;
-    if ts.csi != CSI_ERROR || ts.raw != "────" {
-        return None;
+    /// Return the key when the line is like "--- STD(OUT|ERR): somekey ---"
+    fn stdx_section_key_v1(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter();
+        let first = strings.next()?;
+        if !regex_is_match!(r"^--- STD(OUT|ERR):\s*", &first.raw) {
+            return None;
+        }
+        extract_key_after_crate_name_v1(strings)
     }
-    let ts = strings.find(|ts| ts.raw != " ")?;
-    if !regex_is_match!(r"^STD(OUT|ERR):\s*", &ts.raw) {
-        return None;
+    /// Return the key when the line is like "──── STD(OUT|ERR): cratename some::key"
+    fn stdx_section_key_v2(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter();
+        let ts = strings.next()?;
+        if ts.csi != CSI_ERROR || ts.raw != "────" {
+            return None;
+        }
+        let ts = strings.find(|ts| ts.raw != " ")?;
+        if !regex_is_match!(r"^STD(OUT|ERR):\s*", &ts.raw) {
+            return None;
+        }
+        extract_key_after_crate_name_v2(strings)
+    }
+    /// Return the last test key when the line is like "──── std(out|err)"
+    /// For nextest 0.9.95+ (see https://github.com/Canop/bacon/issues/350)
+    ///
+    fn stdx_section_key_v3(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter().skip_while(|ts| ts.is_blank());
+        let ts = strings.next()?;
+        if ts.csi != CSI_ERROR || (ts.raw != "stderr" && ts.raw != "stdout") {
+            return None;
+        }
+        let ts = strings.next()?;
+        if !ts.is_blank() {
+            return None;
+        }
+        let ts = strings.next()?;
+        if ts.csi != CSI_ERROR || ts.raw != "───" {
+            return None;
+        }
+        if strings.next().is_some() {
+            return None;
+        }
```

Comment:
The output is indented by 4 spaces now -- do you want to unindent it?

Reply:
Wouldn't it be better to keep the indentation, and let users who don't want indentation just pass `--no-output-indent` in the nextest command ?

Reply:
Sure, depends on the UX you'd like to present to users. (Wasn't sure if you do your own formatting that might interfere with the indent.)

The `stdout` and `stderr` lines are slightly different with `--no-output-indent` -- would be worth supporting those too. (I tried to keep them the same, but the slight difference just looks better.)

![image](https://github.com/user-attachments/assets/bea37a13-0c12-4dda-89dc-050431c5117d)

Reply:
Should be OK now